### PR TITLE
PM-11535 Only add password to 2 factor login path if present

### PR DESCRIPTION
**Pull request by dseverns-livefront**
## 🎟️ Tracking
https://bitwarden.atlassian.net/browse/PM-11535

## 📔 Objective
- Prevent crypto error for a SSO user with a MP due to wrongly encoded password being passed to login attempt.
- The way the route was being set  w/o specifying nullable args in the destination: was causing "null" to be decoded and instead of passing `null` to the login function it was attempting to send the resulting string which was incorrect and also was "\<?><?>e" encoded characters.

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed
  issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/twofactorlogin/TwoFactorLoginNavigation.kt [hunk 1]
```diff
@@ -34,8 +34,11 @@ fun NavController.navigateToTwoFactorLogin(
     password: String?,
     navOptions: NavOptions? = null,
 ) {
+    val requiredRoute = "$TWO_FACTOR_LOGIN_PREFIX/$emailAddress"
+    // only add the password if it is not null.
+    val routeOptions = password?.let { "?$PASSWORD=${it.base64UrlEncode()}" } ?: ""
```
Comment (david-livefront) [inline app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/twofactorlogin/TwoFactorLoginNavigation.kt:39]: Is there something inherently wrong with `?$PASSWORD=null`?

Comment (dseverns-livefront) [reply app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/twofactorlogin/TwoFactorLoginNavigation.kt:39]: the arg would then be `args[PASSWORD] == "null"` only in this case "null" gets encoded, then we are passing that string value to the "login( ..., ..., password = theMessIDescribeAbove)"

Comment (david-livefront) [reply app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/twofactorlogin/TwoFactorLoginNavigation.kt:39]: I wonder if we are just missing this?

```kotlin
        arguments = listOf(
            navArgument(EMAIL_ADDRESS) { type = NavType.StringType },
            navArgument(PASSWORD) {
                type = NavType.StringType
                nullable = true
            },
        ),
```

Comment (dseverns-livefront) [reply app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/twofactorlogin/TwoFactorLoginNavigation.kt:39]: you're 100% correct

Comment (dseverns-livefront) [reply app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/twofactorlogin/TwoFactorLoginNavigation.kt:39]: I am going to remove the tests, because A: we are just using the compose navigation here so no need to test it and B: it breaks the null case : {

**Review by david-livefront — APPROVED** [review]
